ui/src/components/common/TradeCard.tsx

```diff
@@ -17,15 +25,19 @@ const TradeCard: React.FC<TradeCardProps> = ({ children, trade }) => {
     const amount = trade.isBuy ? `+ ${trade.qty} ${base}` : `- ${trade.qty} ${base}`;
     const time = new Date(0);
     // timestamp is in nanos with microsecond precision
-    time.setUTCMilliseconds(parseInt(trade.timestamp.substring(0, trade.timestamp.length - 6)))
+    if (isClearedTrade(trade)) {
+        time.setUTCMilliseconds(timestringToInt(trade.eventTimestamp))
```

Comment:
as a tidbit, the ClearedTrade also has a `trade.timestamp` field like the `SettledTradeSide`, but it was in a different format (an ISO string) rather than a UNIX timestamp, so the parsing resulted in the wrong time being displayed. 

The `eventTimestamp` was as expected, but we may wish to normalize this. And I'm also not sure why we have two different timestamp fields to begin with (this was present in DerivativeTradeSide as well I believe)

Reply:
In `ClearedTradeSide`, `eventTimestamp` is what we receive from Exberry, and `timestamp` is a Daml `getTime` for when the trade is novated. I believe in regular `TradeSide`, `timestamp` is the `eventTimestamp` from Exberry.

So I'd say it would be good to make this more consistent, either by having the `timestamp` field in regular trades be `eventTimestamp`, or making the `timestamp` in `ClearedTradeSide` be `timeNovated` or something to that effect. Or both.

Reply:
It might also be worth converting the `eventTimestamp` from Exberry into a Daml `Time` when creating the contracts in the adapter.